### ISSUE TITLE
[DragonAPI 31b] Render issues

### ISSUE THREAD

**Issue by koravel** [ReikaKalseki/Reika_Mods_Issues#3190]
Grass still has black borders. Other things are affected in a similar way. This only happens with DragonAPI included. Related to #2879. 
It is worth to mention that you just ignore a bunch of bugs for years saying that "it does not exist".
But the same bugs are still there. After all the unfixed issues that I found I have one question - do you even run your mods yourself?
Have you heard of testing?

Here is my options file, for better reproducing:
https://pastebin.com/nrYAKUWf

Visual proof:
![2023-05-06_07 55 51](https://user-images.githubusercontent.com/26851016/236600868-d1d4a805-bad5-4fca-872b-d4c6b48b48ca.png)
![2023-05-06_07 56 41](https://user-images.githubusercontent.com/26851016/236600875-72c6ce75-e20a-4eb1-8e68-41eee6415b50.png)

And some more:
![2023-05-06_07 57 35](https://user-images.githubusercontent.com/26851016/236600882-3d6ff14f-a688-41a5-ac9a-b69765a337f3.png)
![2023-05-06_07 57 40](https://user-images.githubusercontent.com/26851016/236600885-b0aeaca7-56f8-4998-a434-184783252133.png)

As well as your lovely porkchop
![2023-05-06_08 04 33](https://user-images.githubusercontent.com/26851016/236601322-d4059b7d-a0d4-4853-9551-bc482e252060.png)

Something new, there is also a slightly burned steak:
![2023-05-06_08 08 21](https://user-images.githubusercontent.com/26851016/236601230-38106cdb-6950-4e19-a468-d289edf0fe5e.png)

>"DAPI does not add random pixels to items."

_Originally posted by @ReikaKalseki in https://github.com/ReikaKalseki/Reika_Mods_Issues/issues/2302#issuecomment-622042121_
            
It seems that is does.

Another proof that minecraft 1.7.10 with forge 10.13.4.1614 does not have burned food (just in case you do not believe that your magnum opus did this):
![2023-05-06_08 13 48](https://user-images.githubusercontent.com/26851016/236601467-777504af-bc4f-48d3-806d-0efa0cd23c90.png)
![2023-05-06_08 13 50](https://user-images.githubusercontent.com/26851016/236601474-d4af24f1-b602-4b15-bc2d-5529eed6b402.png)

Proof that ONLY DragonAPI 31b was installed (no Optifine, it is not here):
![image](https://user-images.githubusercontent.com/26851016/236600921-7b5db1ba-12d8-4ffb-970a-08ad29621557.png)

Also, this render issue potentially related to #3177

**Comment by koravel** [comment]
Update: After playing with video settings I discovered that Mipmap levels affects the grass issue. But, porkchop and steak are still have "burned" texture.  Not affects #3177. Still, it is an issue and disabling Mipmap levels is not a solution, it is just unwanted workaround.

**Comment by Mettymagic** [comment]
dude thinks being a jerk to the mod dev is gonna get their absolutely insignificant complaint fixed 💀 💀 💀 💀 💀

**Comment by koravel** [comment]
@TrueThaumaturge You think calling me a "dude" will make me act the way you want... No, it is not. Behave yourself, you not in a kingergarten. 
But, for all others, let me introduce the problem: Reika is ignoring the same bugs for years, telling that those are not exist, or, just ignoring people telling them some things like in #2302 and #2879, where as result, although bug is clearly here, it is ignored by dev, telling that "it does not exist", "my local env is ok" and so on. This is passive-toxic and annoying behaviour.
Seems like Reika just do not trust? people. Or that Reika thinks he is very intelligent an sentient being that do not care even when there is a problem. "DAPI does not add random pixels to items." - that says exactly what I was talking about, "not exist", "don't care", "go fk yourself it's not my problem, it is yours" and so on. Or, it can tell that Reika, for the same reason, just don't test and play their own mods, thinking he is so cool and godlike that this is not needed. 

At least it looks like it and causes such a reaction. Welcome anyone if you agree or do not agree with me, that bugs were ignored for years for some weird reasons or for no reason at all. 

@ReikaKalseki I want to hear from you what exactly is the problem, why do you ignore people when they telling you that there is a bug? Honestly, you've created good stuff, aside from that you talking to people like they are the piece of shit. You created and published these mods for people, as I think. So you make it for them. Why after all you treat em like that? They want to help, providing you with valuable info about problems in your mods, even with inconsistent form it still valuable, and you be like "go fk yourself". This is just not right.

**Comment by brandyyn** [comment]
I had the same problem that I reported a few years ago, I fixed it by installing optifine. #2302

actually, I saw you noted my post in your reply; but i'm shocked it's still not fixed- I just dealt with it for a while until I got optifine working in my pack and then I forgot about it lol.

**Comment by TrueThaumaturge** [comment]
@koravel for US https://www.nimh.nih.gov/health/find-help https://www.cdc.gov/mentalhealth/tools-resources/index.htm 
for CA https://www.canada.ca/en/public-health/services/mental-health-services/mental-health-get-help.html https://kidshelpphone.ca/
for EU https://www.mhe-sme.org/ https://checkpointorg.com/global/

**Comment by ReikaKalseki** [comment]
@koravel Your allegation of "ignoring the issue", and this post in particular, is a result of my having been modding other games (and thus not touching MC since Jan 2022) until very recently, and still not having gotten around to looking at the issue tracker.

More generally, this issue, no matter how real it is, is **not** a serious issue and as such not worth sinking large amounts of time into when there are many other issues or even new features which are much more worth the dev time.

On top of that, you appear to not be aware of the significant influence that *presentation* of an issue has on both what I think the user is reporting as well as what I test (codewise or ingame) to verify its reality or work out a cause. This issue was initially reported as DAPI *literally* modifying texture data, which it did not do then and does not do now, hence the "random pixels" comment you like to keep harping on. My guess for what is *actually* going on is a GL state (blending in particular) issue, in that these sprites contain at-or-near-full-transparency pixels in those locations that are normally not visible but become visible due to a change in those GL flags. As for why that change exists, I can guess it has something to do with one of many rendering changes I made over the years, with one leading suspect being a 2016-era change where I changed the glAlphaFunc in all `Tessellator` calls. Fortunately for you, that actually has been changed in the next version, so this issue *might* already be fixed. If not, however, I am not going to be spending large amounts of time either trying to work out the real cause or trying to resolve it without breaking desired rendering behavior; as I allude to above, if your biggest complaint is that pork chops have a couple black pixels on them, that is a pretty glowing commendation.

Finally, I am going to point out that your response here *is* very disproportionate to the severity of this issue, even before you start alleging - baselessly - that I am just "telling people to F#$% off" or the like.
